Author: Chris Rogers <rogersc@ainfosec.com>
Date:   Tue Apr 18 16:23:47 2017 -0400

Support swapping isos in virtual cd drive

while guest is on. Make sure we set readonly flags for blockdev
in qemu when set to 'preserved'. Attempting to open xvdc fails
because open flags have writable set.

OXT-1068

Signed-off-by: Chris Rogers <rogersc@ainfosec.com>

--- a/blockdev.c
+++ b/blockdev.c
@@ -2683,6 +2683,7 @@ void qmp_blockdev_change_medium(bool has
 
     switch (read_only) {
     case BLOCKDEV_CHANGE_READ_ONLY_MODE_RETAIN:
+        bdrv_flags &= ~BDRV_O_RDWR;
         break;
 
     case BLOCKDEV_CHANGE_READ_ONLY_MODE_READ_ONLY:
